vault/token_store.go

```diff
@@ -300,6 +337,27 @@ func (ts *TokenStore) rootToken() (*TokenEntry, error) {
 	return te, nil
 }
 
+// CreateAccessorID is used to create an identifier for the token ID.
+// An storage index, mapping the accessor ID to the token ID is also created.
+func (ts *TokenStore) createAccessorID(entry *TokenEntry) error {
+	defer metrics.MeasureSince([]string{"token", "createAccessorID"}, time.Now())
+
+	// Create a random accessor ID
+	accessorUUID, err := uuid.GenerateUUID()
+	if err != nil {
+		return err
+	}
+	entry.AccessorID = accessorUUID
+
+	// Create index entry, mapping the Accessor ID to the Token ID
+	path := lookupPrefix + ts.SaltID(entry.AccessorID)
```

Comment:
I think we should use a new prefix for this, to keep things logically separated. Makes things easier if we ever need to e.g. iterate over the entries to perform an upgrade (for tokens or accessors).